src/downloader/headers/preverified_hashes_config.rs

```diff
@@ -0,0 +1,37 @@
+use std::collections::HashMap;
+
+/// The preverified hashes is a list of known precomputed hashes of every 192-th block in the chain:
+///
+/// hash(0), hash(192), hash(384), hash(576), ...
+///
+/// The preverified hashes are copied from:
+/// https://github.com/ledgerwatch/erigon/blob/devel/turbo/stages/headerdownload/preverified_hashes_mainnet.go
+/// https://github.com/ledgerwatch/erigon/blob/devel/turbo/stages/headerdownload/preverified_hashes_ropsten.go
+pub struct PreverifiedHashesConfig {
+    pub hashes: Vec<ethereum_types::H256>,
+}
+
+impl PreverifiedHashesConfig {
+    pub fn new(chain_name: &str) -> anyhow::Result<Self> {
+        let config_text = match chain_name {
+            "mainnet" => include_str!("preferified_hashes_mainnet.toml"),
+            "ropsten" => include_str!("preferified_hashes_ropsten.toml"),
+            _ => anyhow::bail!("unsupported chain"),
+        };
+        let config: HashMap<String, Vec<String>> = toml::from_str(config_text)?;
+        Ok(PreverifiedHashesConfig {
+            hashes: PreverifiedHashesConfig::parse_hashes(&config["hashes"]),
```

Comment:
Must deserialize directly into final data model using `toml::from_str`, unless there is a good reason not to.

Reply:
Is there a way to do it without prepending "0x" to all the data values?

Reply:
https://github.com/akula-bft/akula/blob/master/src/models/chainspec.rs#L44
https://github.com/akula-bft/akula/blob/master/src/models/chainspec.rs#L158

In your case, you could make a deserializer that deserializes into String, and then parses as H256.

It will allocate, so when you have time, you could instead deserialize through a wrapper type, like `UnprefixedH256`, that would simply omit [this](https://github.com/paritytech/parity-common/blob/master/primitive-types/impls/serde/src/serialize.rs#L96) check.